gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/keyedgrouping/functions/BuildSuperElementFromTuple.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.flink.api.java.tuple.Tuple;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.gradoop.common.model.api.entities.Element;
+import org.gradoop.common.model.impl.properties.Properties;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.api.functions.AggregateFunction;
+import org.gradoop.flink.model.api.functions.KeyFunction;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Build the final super-elements from the internal tuple-based representation.
+ *
+ * @param <T> The input tuple type.
+ * @param <E> The final element type.
+ */
+abstract class BuildSuperElementFromTuple<T extends Tuple, E extends Element>
+  implements MapFunction<T, E>, ResultTypeQueryable<E> {
+
+  /**
+   * The data offset for tuples. Grouping keys and aggregate values are expected to start at this
+   * index.
+   */
+  private final int tupleDataOffset;
+
+  /**
+   * The grouping key functions..
+   */
+  private final List<KeyFunction<E, ?>> keyFunctions;
+
+  /**
+   * The aggregate functions for this element type.
+   */
+  private final List<AggregateFunction> aggregateFunctions;
+
+  /**
+   * Initialize this function.
+   *
+   * @param tupleDataOffset    The number of reserved fields in the tuple.
+   * @param groupingKeys       The grouping key functions.
+   * @param aggregateFunctions The aggregate functions.
+   */
+  protected BuildSuperElementFromTuple(int tupleDataOffset,
+    List<KeyFunction<E, ?>> groupingKeys,
+    List<AggregateFunction> aggregateFunctions) {
+    this.tupleDataOffset = tupleDataOffset;
+    this.keyFunctions = Objects.requireNonNull(groupingKeys);
+    this.aggregateFunctions = Objects.requireNonNull(aggregateFunctions);
+  }
+
+  /**
+   * Calculate the final aggregate values by calling post-processing functions and set them
+   * to the element. Also set grouping keys as properties on the element.
+   *
+   * @param element   The element to store the properties.
+   * @param tupleData The internal tuple-based representation of the element.
+   * @return The final element with all set properties.
+   */
+  protected E setAggregatePropertiesAndKeys(E element, T tupleData) {
+    element.setLabel("");
+    element.setProperties(Properties.create());
+    // Set grouping keys.
+    for (int i = 0; i < keyFunctions.size(); i++) {
+      final Object groupingKey = tupleData.getField(tupleDataOffset + i);
+      keyFunctions.get(i).addKeyToElement(element, groupingKey);
+    }
+    // Calculate aggregate values and set them.
+    for (int i = 0; i < aggregateFunctions.size(); i++) {
+      final AggregateFunction function = aggregateFunctions.get(i);
+      final PropertyValue postAggregateValue = function.postAggregate(
+        tupleData.getField(tupleDataOffset + keyFunctions.size() + i));
+      if (postAggregateValue != null) {
+        element.setProperty(function.getAggregatePropertyKey(), postAggregateValue);
```

Comment:
You forgot to set default aggregates. You can maybe use `AggregateUtil.getDefaultAggregate` for this. Default values are set after `postAggregate`.